python_modules/libraries/dagster-celery-k8s/dagster_celery_k8s/executor.py

```diff
@@ -368,6 +387,11 @@ def _execute_step_k8s_job(
             labels["dagster/code-location"] = (
                 dagster_run.external_job_origin.repository_origin.code_location_origin.location_name
             )
+        per_op_override = per_step_k8s_config.get(step_key, {})
+        # NOTE per_op_override will take precedence
+        user_defined_k8s_config = UserDefinedDagsterK8sConfig.from_dict(
+            deep_merge_dicts(user_defined_k8s_config.to_dict(), per_op_override)
+        )
```

Comment:
we have some existing code/patterns for merging UserDefinedDagsterK8sConfig objects that we should use here:

```
tag_container_context = K8sContainerContext(run_k8s_config=user_defined_k8s_config)
executor_config_container_context = K8sContainerContext(run_k8s_config=UserDefinedDagsterK8sConfig.from_dict(per_op_override))

merged_user_defined_k8s_config = tag_container_context.merge(executor_config_container_context).run_k8s_config
```

Reply:
would it give same behaviour in terms of override precedence?

Reply:
yes, the input to merge gets precedence.